lib/install/management/rbac.go

```diff
@@ -207,6 +220,89 @@ func (am *AuthzManager) setupPermissions(ctx context.Context) ([]RBACResourcePer
 	return resourcePermissions, nil
 }
 
+func (am *AuthzManager) isPrincipalAnAdministrator(ctx context.Context) (bool, error) {
+	// Check if the principal belongs to the Administrators group
+	res, err := am.principalBelongsToGroup(ctx, "Administrators")
+	if err != nil {
+		return false, err
+	}
+
+	if res {
+		return res, nil
+	}
+
+	// Check if the principal has an Admin Role
+	res, err = am.principalHasRole(ctx, "Admin")
+	if err != nil {
+		return false, err
+	}
+
+	return res, nil
+}
+
+func (am *AuthzManager) principalBelongsToGroup(ctx context.Context, group string) (bool, error) {
+	ref := *am.client.ServiceContent.UserDirectory
+
+	components := strings.Split(am.principal, "@")
+	if len(components) < 2 {
```

Comment:
I think this will fail on ESX (although there may be another check that stops us getting here).
I do not know if it's legitimate to have a non-domain user for VC, but the RetrieveUserGroups permits an "" domain and if you list the domains for a VC then "" is in the list.
Can we allow a missing domain to support `root` on ESX for example? If so then we'd just conditionally assign to domain based on `len(components)`.